editor/post-sticky/check.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+import { flowRight } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { withAPIData } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { getCurrentPostType } from '../selectors';
+
+export function PostStickyCheck( { postType, children, user } ) {
```

Comment:
To avoid duplicating the "support sticky" check, I extracted it to its own component. I could reuse this approach for all the other sidebar panels.

Reply:
This works well.

Otherwise, I could imagine the following:

```js
// new HoC
const withUserPrivileges = ( predicate ) =>
  flowRight(
    withAPIData( constant( { user: '/wp/v2…' } ) ),
    withRenderIf( predicate )
  );

// consumer
import withUserPrivilege from …;
const applyPostStickyCheck = withUserPrivileges( ( { postType, user } ) =>
  postType !== 'post' ||
    ! user.data ||
    ! user.data.capabilities.publish_posts ||
    ! user.data.capabilities.edit_others_posts
);

export default applyPostStickyCheck( MyComponent );
```

Reply:
This could obfuscate component render logic, though. Right now most HoCs only deal with prop injection and side effects (like click-outside), not affecting _when_ things should be rendered. Which is why the current way seems appropriate. 👌